# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


In [8]:
import numpy as np
from scipy import stats

# Дані з умови
time_to_purchase = np.array([128, 135, 121, 142, 126, 151, 123])
mu0 = 123.7   # середній час до покупки
alpha = 0.10  # рівень значущості

**Гіпотези**

H₀: μ = 123,7 (середній час до покупки залишився на рівні 123,7 хв)  
H₁: μ > 123,7 (середній час збільшився)  
Тип тесту: правосторонній (one-tailed)

In [9]:
# Обчислимо вибіркову статистику
x_bar = np.mean(time_to_purchase)
s = np.std(time_to_purchase, ddof=1)   # стандартне відхилення з n-1
n = len(time_to_purchase)

Оскільки маємо малу вибірку (n=7), то використаємо t-тест

In [10]:
# Обчислимо t-статистику
t_stat = (x_bar - mu0) / (s / np.sqrt(n))

# Критичне значення для правостороннього тесту
df = n - 1    # ступені свободи
t_crit = stats.t.ppf(1 - alpha, df)

# p-value (правосторонній тест)
p_value = 1 - stats.t.cdf(t_stat, df)


In [7]:
print(f"Вибіркове середнє: {x_bar:.2f}")
print(f"Вибіркове стандартне відхилення: {s:.2f}")
print(f"t-статистика: {t_stat:.3f}")
print(f"Критичне значення (df={df}): {t_crit:.3f}")
print(f"p-value: {p_value:.4f}")

# Висновок
if t_stat > t_crit and p_value < alpha:
    print(f"Відхиляємо H₀: середній час до покупки статистично значуще зріс після редизайну (α={alpha})")
else:
    print(f"Не відхиляємо H₀: недостатньо доказів, що середній час до покупки збільшився (α={alpha})")


Вибіркове середнє: 132.29
Вибіркове стандартне відхилення: 10.98
t-статистика: 2.069
Критичне значення (df=6): 1.440
p-value: 0.0420
Відхиляємо H₀: середній час до покупки статистично значуще зріс після редизайну (α=0.1)


𝑡 ≈ 2.07  
p ≈ 0.042  

При α = 0.10: p < 0.10 → відхиляємо H₀ → є докази, що редизайн збільшив час до покупки.

При α = 0.05: p < 0.05 → знову відхиляємо H₀ → висновок такий самий.

**Висновок**  
Редизайн сайту статистично значуще подовжив час до покупки користувачів.  
Рішення не змінюється ні при α = 0.10, ні при α = 0.05.

**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

In [17]:
import statsmodels.api as sm
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

count = 311    # кількість "успіхів"
nobs = 1200    # обсяг вибірки
p0 = 0.23      # гіпотетична пропорція (H0)

# z-тест для пропорції
stat, pval = proportions_ztest(count, nobs, p0, alternative='larger')

# довірчий інтервал (1 - alpha) = 97%
confint = proportion_confint(count, nobs, alpha=0.03, method='normal')

print("🔹 Тест для однієї пропорції")
print(f"Z-статистика = {stat:.3f}")
print(f"p-value = {pval:.4f}")
print(f"{(1-0.03)*100:.0f}% довірчий інтервал: {confint}")

🔹 Тест для однієї пропорції
Z-статистика = 2.306
p-value = 0.0106
97% довірчий інтервал: (0.23171700302179205, 0.28661633031154127)


**Висновки**  
Оскільки  p-value=0.0106 < 0.03, відхиляємо  H0 . Є статистично значущі докази,що частка дорослих, які впізнають логотип, зросла після кампанії і тепер більша за 23%

97% довірчий інтервал (0.236, 0.280)
Це означає, що з імовірністю 97% справжня частка впізнаваності лежить у межах від 23.6% до 28.0%.
